src/BlockTypes/ProductGalleryLargeImage.php

```diff
@@ -102,36 +99,66 @@ function( $carry, $key ) use ( $directives ) {
+	 *
+	 * @param array $context The block context.
+	 * @param int   $product_id The product id.
 	 *
-	 * @param string $image_html The image HTML.
 	 * @return array
 	 */
-	private function get_html_with_interactivity( $image_html ) {
-		$context = array(
-			'woocommerce' => array(
-				'styles' => array(
-					'transform'        => 'scale(1.0)',
-					'transform-origin' => '',
-				),
-			),
-		);
+	private function get_main_images_html( $context, $product_id ) {
+		$attributes = array(
+			'data-wc-bind--hidden' => '!selectors.woocommerce.isSelected',
+			'hidden'               => true,
+			'class'                => 'wc-block-woocommerce-product-gallery-large-image__image',
 
-		$directives = array(
-			'data-wc-on--mousemove'  => 'actions.woocommerce.handleMouseMove',
-			'data-wc-on--mouseleave' => 'actions.woocommerce.handleMouseLeave',
-			'data-wc-context'        => wp_json_encode( $context, JSON_NUMERIC_CHECK ),
 		);
 
-		$image_html_processor = new \WP_HTML_Tag_Processor( $image_html );
-		$image_html_processor->next_tag( 'img' );
-		$image_html_processor->add_class( 'wc-block-woocommerce-product-gallery-large-image__image--hoverZoom' );
-		$image_html_processor->set_attribute( 'data-wc-bind--style', 'selectors.woocommerce.styles' );
-		$image_html = $image_html_processor->get_updated_html();
+		if ( $context['hoverZoom'] ) {
+			$attributes['class']              .= ' wc-block-woocommerce-product-gallery-large-image__image--hoverZoom';
+			$attributes['data-wc-bind--style'] = 'selectors.woocommerce.styles';
+		}
 
-		return array(
-			$directives,
-			$image_html,
+		$main_images = ProductGalleryUtils::get_product_gallery_images(
+			$product_id,
+			'full',
+			$attributes
 		);
+
+		$visible_main_image           = array_shift( $main_images );
+		$visible_main_image_processor = new \WP_HTML_Tag_Processor( $visible_main_image );
+		$visible_main_image_processor->next_tag();
+		$visible_main_image_processor->remove_attribute( 'hidden' );
+		$visible_main_image = $visible_main_image_processor->get_updated_html();
+
+		return array( $visible_main_image, $main_images );
+
+	}
+
+	/**
+	 * Get directives for the hover zoom.
+	 *
+	 * @param array $block_context The block context.
+	 *
+	 * @return array
+	 */
+	private function get_directives( $block_context ) {
+		if ( ! $block_context['hoverZoom'] ) {
+			return array();
+		}
+			$context = array(
+				'woocommerce' => array(
+					'styles' => array(
+						'transform'        => 'scale(1.0)',
+						'transform-origin' => '',
+					),
+				),
+			);
+
+			return array(
+				'data-wc-on--mousemove'  => 'actions.woocommerce.handleMouseMove',
+				'data-wc-on--mouseleave' => 'actions.woocommerce.handleMouseLeave',
+				'data-wc-context'        => wp_json_encode( $context, JSON_NUMERIC_CHECK ),
+			);
 	}
```

Comment:
It looks like there are some spaces left here after the IF condition:
```suggestion
	private function get_directives( $block_context ) {
		if ( ! $block_context['hoverZoom'] ) {
			return array();
		}
	    $context = array(
		    'woocommerce' => array(
			    'styles' => array(
				    'transform'        => 'scale(1.0)',
				    'transform-origin' => '',
			    ),
		    ),
	    );
    
	    return array(
		    'data-wc-on--mousemove'  => 'actions.woocommerce.handleMouseMove',
		    'data-wc-on--mouseleave' => 'actions.woocommerce.handleMouseLeave',
		    'data-wc-context'        => wp_json_encode( $context, JSON_NUMERIC_CHECK ),
	    );
	}
```